### PR TITLE
Slack sign in

### PR DESCRIPTION
**Pull request by musialik**

### CODE REVIEW

File: web/router.ex [hunk 1]
```diff
@@ -17,6 +17,8 @@ defmodule Integrator.Router do
     pipe_through :browser # Use the default browser stack
 
     get "/", PageController, :index
+    get "/auth", AuthController, :index
+    get "/auth/callback", AuthController, :callback
```
Comment (norbertnytko) [inline web/router.ex:21]: shouldn't this be`get "/auth/slack/callback", AuthController, :callback` ?

File: web/controllers/auth_controller.ex [hunk 2]
```diff
@@ -0,0 +1,18 @@
+defmodule Integrator.AuthController do
+  use Integrator.Web, :controller
+
+  def index(conn, _params) do
+    redirect conn, external: Slack.authorize_url!(scope: "identity.basic")
+  end
+
+  def callback(conn, params) do
```
Comment (norbertnytko) [inline web/controllers/auth_controller.ex]: Any idea how to handle failure?

File: mix.lock [hunk 3]
```diff
@@ -1,24 +1,31 @@
...
-  "plug": {:hex, :plug, "1.3.5", "7503bfcd7091df2a9761ef8cecea666d1f2cc454cbbaf0afa0b6e259203b7031", [:mix], [{:cowboy, "~> 1.0.1 or ~> 1.1", [hex: :cowboy, repo: "hexpm", optional: true]}, {:mime, "~> 1.0", [hex: :mime, repo: "hexpm", optional: false]}], "hexpm"},
-  "poison": {:hex, :poison, "2.2.0", "4763b69a8a77bd77d26f477d196428b741261a761257ff1cf92753a0d4d24a63", [:mix], [], "hexpm"},
-  "poolboy": {:hex, :poolboy, "1.5.1", "6b46163901cfd0a1b43d692657ed9d7e599853b3b21b95ae5ae0a777cf9b6ca8", [:rebar], [], "hexpm"},
-  "postgrex": {:hex, :postgrex, "0.13.2", "2b88168fc6a5456a27bfb54ccf0ba4025d274841a7a3af5e5deb1b755d95154e", [:mix], [{:connection, "~> 1.0", [hex: :connection, repo: "hexpm", optional: false]}, {:db_connection, "~> 1.1", [hex: :db_connection, repo: "hexpm", optional: false]}, {:decimal, "~> 1.0", [hex: :decimal, repo: "hexpm", optional: false]}], "hexpm"},
-  "ranch": {:hex, :ranch, "1.3.2", "e4965a144dc9fbe70e5c077c65e73c57165416a901bd02ea899cfd95aa890986", [:rebar3], [], "hexpm"}}
+%{"bunt": {:hex, :bunt, "0.2.0", "951c6e801e8b1d2cbe58ebbd3e616a869061ddadcc4863d0a2182541acae9a38", [:mix], []},
```
Comment (norbertnytko) [inline mix.lock]: For some reason after build this adds me `"hexpm"` at the end of line.

Comment (musialik) [reply mix.lock]: maybe we have different versions of hex?

File: web/controllers/auth_controller.ex (hunk 2 above)
Comment (musialik) [reply web/controllers/auth_controller.ex]: I guess the params are different in this case, so we could just use pattern matching and define two functions

File: web/router.ex (hunk 1 above)
Comment (musialik) [reply web/router.ex:21]: Do we need to worry about other auth providers?

**Review by netgurusupporter — APPROVED** [review]
LGTM

**Pull request merged**